Ductus.FluentDocker/Executors/ProcessExecutor.cs

```diff
@@ -30,64 +31,37 @@ public ProcessExecutor(string command, string arguments, string workingdir = nul
 
     public IDictionary<string, string> Env { get; } = new Dictionary<string, string>();
 
-    public CommandResponse<TE> Execute(
-    DataReceivedEventHandler outputDataReceived = null,
-    DataReceivedEventHandler errorDataReceived = null)
+    public CommandResponse<TE> Execute([CallerMemberName] string caller = "")
     {
-      var startInfo = new ProcessStartInfo
-      {
-        CreateNoWindow = true,
-        RedirectStandardOutput = true,
-        RedirectStandardError = true,
-        UseShellExecute = false,
-        Arguments = _arguments,
-        FileName = _command,
-        WorkingDirectory = _workingdir
-      };
+      Logger.Log($"cmd: {_command} - arg: {_arguments}");
 
-      if (0 != Env.Count)
-        foreach (var key in Env.Keys)
-        {
-#if COREFX
-          startInfo.Environment[key] = Env[key];
-#else
-          startInfo.EnvironmentVariables[key] = Env[key];
-#endif
-        }
+      var pm = ProcessManagerContext.ProcessManager;
```

Comment:
Could'n we use a instance based approach for a `ProcessManager` instead of a static one and thus avoid thread variables (? - haven't checked what you use them for). 

like use(pm = ...) {
 fd.UsePm(pm).
  ...
} 

when the pm context exits, all commands for compse and docker services, etc. will fail (with e.g. missing pm (name xyz)). Or at-least a non static approach?

Reply:
I have simplified what is going on here This is needed however its how you access the AmbientContext. If you look under the hood of that context its getting a thread variable. Like I said I feel a bit dirty.

Reply:
I think a better approach is to skip the `AmbientContext` and just pass the `ProcessManager` into the commands instead - just use a static default if not provided?